Typings cleanup

alexeagle opened the pull request:
We had our own fork due to some custom typings, but these have been pushed upstream.

alexeagle commented:
Something is fishy. In angular's fork we have  "zone" (https://github.com/angular/DefinitelyTyped/blob/master/zone/zone.d.ts). The one on DefinitelyTyped has "zone.js" (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/zone.js/zone.js.d.ts#L6)

This PR should have failed on Travis because the zone typings are missing.
the log has under npm install:
```
>> written 7 files:
    - angular-protractor/angular-protractor.d.ts
    - es6-shim/es6-shim.d.ts
    - hammerjs/hammerjs.d.ts
    - jasmine/jasmine.d.ts
    - node/node.d.ts
    - selenium-webdriver/selenium-webdriver.d.ts
    - zone/zone.d.ts
```

Where did it find `zone.d.ts`?? Did it come from the Travis cache even though this PR should bust the cache? Fishy.

alexeagle commented:
Hmm, I think it's because we give a SHA, and that can be resolved against a repo that doesn't have that in the tree anywhere: this works
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/31e7317c9a0793857109236ef7c7f223305a8aa9
even though that change only appeared on the angular fork. So maybe it's legit that Travis passed.
I'll update the PR though'

alexeagle commented:
cc @vikerman

rkirov commented on modules/angular1_router/tsd.json:
```diff
@@ -1,12 +1,12 @@
 {
   "version": "v4",
-  "repo": "angular/DefinitelyTyped",
+  "repo": "DefinitelyTyped/DefinitelyTyped",
```
no more forks!

rkirov commented on modules/angular2/manual_typings/globals.d.ts:
```diff
@@ -1,7 +1,52 @@
+ * Subset of es6-shim typings.
+ * Angular should not require use of ES6 runtime but some API usages are already present.
+ * See https://github.com/angular/angular/issues/5242
+ * TODO(alexeagle): remove methods below which may not be present in targeted browser
```
Not sure what you are trying to say with that TODO. We are targetting IE9 and it doesn't have any of those.

alexeagle replied:
so maybe the TODO is remove all of these - but I don't imagine we will really do that?

alexeagle replied:
@mhevery FYI

rkirov commented on modules/playground/e2e_test/material/progress_linear_spec.ts:
```diff
@@ -14,10 +14,10 @@ describe('md-progress-linear', function() {
 
     incrementButton.click();
-    expect(progressBar.getAttribute('aria-valuenow')).toBeGreaterThan(initialValue);
+    expect(progressBar.getAttribute('aria-valuenow')).toBeGreaterThan(<any>initialValue);
```
what happened here.

alexeagle replied:
Turns out this was not type-checked before, our fork of DT did not have the right propagation of the type to the argument of these methods like toBeGreaterThan.

rkirov commented:
looks good, make lint pass

alexeagle committed:
chore(deps): update ts2dart and zone.js

alexeagle committed:
chore(typings): use mainline DefinitelyTyped repo rather than a fork.

The upstream Jasmine typings don't define a type for the global
object with Jasmine methods polluting it, so just use any.

Also zone.js has a different name upstream.

alexeagle committed:
fix(typings): Don't expose typing dependencies to users.

This resolves Duplicate Identifier issues seen by many users,
at the expense of more typings installation required in some
cases.

Removes the quickstart hack of placing all needed dependencies
typings files in our distribution. Removes dependencies on
nodejs from angular2/core.

Fixes #5973
Fixes #5807
Fixes #6266

Angular now depends on es6-promise and es6-collections
(and a handful of manual typings) rather than all of es6-shim.

Fixes #5242

We previously had an undocumented breaking change, this is now
documented in this commit.

Fixes #6817

BREAKING CHANGE:

Transitive typings are no longer included in the distribution.
You may need to install typings in your project using
http://github.com/typings/typings

Users now must rely on getting typings from:
- one of the peerDependencies, such as rxjs, which exposes
  typings via the moduleResolution=node mechanism.
  (see https://github.com/Microsoft/TypeScript/wiki/Typings-for-npm-packages)
  This happens automatically.
- Using --target ES5 now requires manual installation of
  es6-promise and es6-collections typings.
- Using some angular APIs may introduce a dependency on eg. nodejs
  or jasmine, and those typings need manual installation as well.

btford commented:
Is this ready to merge?

alexeagle commented:
not ready to merge yet

alexeagle commented:
Reviewed in person with @mhevery

alexeagle commented:
cc @blakeembrey who helped with the design for this - thank you!

mary-poppins commented:
Merging PR #6267 on behalf of @IgorMinar to branch [presubmit-IgorMinar-pr-6267](https://github.com/angular/angular/tree/presubmit-IgorMinar-pr-6267).
